### PR TITLE
Fix/tao 6637/log 500 exceptions

### PR DESCRIPTION
**Pull request by SergiiTao**

### CODE REVIEW

File: scripts/update/Updater.php [hunk 1]
```diff
@@ -875,6 +875,9 @@ public function update($initialVersion) {
             $this->setVersion('21.5.0');
         }
 
+<<<<<<< HEAD
+        $this->skip('21.5.0', '23.0.0');
```
Comment (siwane) [inline scripts/update/Updater.php]: *biiiip*

File: models/classes/mvc/error/ExceptionInterpretor.php [hunk 2]
```diff
@@ -107,7 +106,7 @@ protected function interpretError() {
     }
     
     public function getTrace() {
-        return $this->trace;
+        return $this->exception ? $this->exception->getMessage() : '';
```
Comment (Skyld) [inline models/classes/mvc/error/ExceptionInterpretor.php]: Why would a method called `getTrace` return an exception message? Please either rename the method to fulfill it's purpose or return the trace

File: models/classes/mvc/error/ExceptionInterpretor.php [hunk 3]
```diff
@@ -130,11 +129,11 @@ public function getResponse() {
         $class = $this->getResponseClassName();
         /*@var $response ResponseAbstract */
         $response = new $class;
-        $response->setServiceLocator($this->getServiceLocator())
-                ->setException($this->exception)
-                ->setHttpCode($this->returnHttpCode)
-                ->trace($this->trace);
+        $response->setServiceLocator($this->getServiceLocator());
```
Comment (Skyld) [inline models/classes/mvc/error/ExceptionInterpretor.php]: This seems weird to me. How is it ensured that exception was set here through `setException` method? Why do we need the `getResponseClassName` method? It should be namespace prefixed before already. This should be refactored.

File: models/classes/mvc/error/ExceptionInterpretor.php [hunk 4]
```diff
@@ -74,7 +74,6 @@ public function setException(Exception $exception){
      * and http status to return
      */
     protected function interpretError() {
```
Comment (Skyld) [inline models/classes/mvc/error/ExceptionInterpretor.php]: Please fix all the conventional errors in the class

File: models/classes/mvc/error/ResponseAbstract.php [hunk 5]
```diff
@@ -129,9 +129,11 @@ public function send()
     /**
```
Comment (Skyld) [inline models/classes/mvc/error/ResponseAbstract.php]: Please fix all the conventional errors in this class

**Comment by Skyld** [comment]
Please resolve conflicts with develop

File: models/classes/mvc/error/ExceptionInterpretor.php (hunk 2 above)
Comment (SergiiTao) [reply models/classes/mvc/error/ExceptionInterpretor.php]: had the same question.
it should be used in one place only, but have no idea about forks.
if you want to rename it we should depreacte it, create the new one, wait some time...

File: models/classes/mvc/error/ExceptionInterpretor.php (hunk 3 above)
Comment (SergiiTao) [reply models/classes/mvc/error/ExceptionInterpretor.php]: it is weird indeed. but here `$response->setException($this->exception)` we're using the setter again.
it means (should mean) that this is not important value and if we set it to null nothing bad happens.

File: models/classes/mvc/error/ExceptionInterpretor.php (hunk 4 above)
Comment (SergiiTao) [reply models/classes/mvc/error/ExceptionInterpretor.php]: I can do it as a separate PR. To make a life of reviewer easier ;)

File: models/classes/mvc/error/ExceptionInterpretor.php (hunk 2 above)
Comment (Skyld) [reply models/classes/mvc/error/ExceptionInterpretor.php]: Okay I see it was not you, leave it as it is then, it's not in scope now.

File: models/classes/mvc/error/ExceptionInterpretor.php (hunk 3 above)
Comment (Skyld) [reply models/classes/mvc/error/ExceptionInterpretor.php]: Where? You just removed the `setException` call from here. What do you mean?

Comment (Skyld) [reply models/classes/mvc/error/ExceptionInterpretor.php]: Ignore it, as discussed

**Pull request merged**